superset-frontend/src/SqlLab/components/ResultSet.jsx

```diff
@@ -258,19 +258,32 @@ export default class ResultSet extends React.PureComponent {
       }
     }
     if (query.cached || (query.state === 'success' && !query.results)) {
-      return (
-        <Button
-          bsSize="sm"
-          className="fetch"
-          bsStyle="primary"
-          onClick={this.reFetchQueryResults.bind(this, {
-            ...query,
-            isDataPreview: true,
-          })}
-        >
-          {t('Fetch data preview')}
-        </Button>
-      );
+      if (query.isDataPreview) {
+        return (
+          <Button
+            bsSize="sm"
+            className="fetch"
+            bsStyle="primary"
+            onClick={this.reFetchQueryResults.bind(this, {
```

Comment:
https://www.freecodecamp.org/news/the-best-way-to-bind-event-handlers-in-react-282db2cf1530/

In Superset code base, usually we bind in constructor.

Reply:
got it. It was bound like this before, but I'll move the binding in both places to the constructor